### ISSUE TITLE
Resume Training

### ISSUE THREAD

**Issue by tiredsoul21** [karpathy/nanoGPT#467]
https://github.com/karpathy/nanoGPT/blob/325be85d9be8c81b436728a420e85796c57dba7e/train.py#L106

In my implementation of the code, I modified this line to incorporate the iteration into the seed.  I suspect that if you resume training multiple times, the random seed may draw the same training sets and in the same order.  Over many iterations of the dataset this may be lost, but it may cause a pattern of validating against the same sets as well, which has higher consequences.

**Comment by VatsaDev** [comment]
> the random seed may draw the same training sets and in the same order

That would be amazingly rare wouldn't it?

**Comment by tiredsoul21** [comment]
I would say given the setup described no.  Sampling out to be deterministic over multiple resumes...right?

**Comment by VatsaDev** [comment]
Oh Wait you're using a manual seed, sry I misread manual seed, well that should load the batches in a certain order, so yes it could do multiple epochs in the same order, You could remove the seed for true randomness, or do the one line change you made, It would be a quick PR if Andrej sees it, but its not already a feature probably because its an uncommon scenario.

Have you experienced it yet by the way? If the batches are being loaded in the same order, shouldn't it start showing as a quick drop on the loss graph, which slowly rises again?
